client/src/components/ProTx/components/charts/MaltreatmentTypesPlot.js

```diff
@@ -0,0 +1,329 @@
+import './MaltreatmentTypesPlot.css';
+
+// Set this to true to inspect the component data in a tabular view.
+// This will hide the chart rendering.
+const debugState = false;
+
+function MaltreatmentTypesPlot({
+  mapType,
+  geography,
+  maltreatmentTypes,
+  observedFeature,
+  year,
+  selectedGeographicFeature,
+  data
+}) {
+  // Define Data Marshalling Methods.
+
+  const getMaltreatmentTypeNames = maltreatmentTypeCodes => {
+    const updatedMaltreatmentTypesList = [];
+    if (maltreatmentTypeCodes.length === 0) {
+      return ['None'];
+    }
+    for (let i = 0; i < maltreatmentTypeCodes.length; i += 1) {
+      for (let j = 0; j < maltreatmentMeta.length; j += 1) {
+        if (maltreatmentTypeCodes[i] === maltreatmentMeta[j].field) {
+          updatedMaltreatmentTypesList.push(maltreatmentMeta[j].name);
+        }
+      }
+    }
+    return updatedMaltreatmentTypesList;
+  };
+
+  const getMaltreatmentTypesDataObject = (codeArray, nameArray, valueArray) => {
+    const newMaltreatmentDataObject = [];
+    for (let i = 0; i < codeArray.length; i += 1) {
+      const dataObject = {};
+      dataObject.code = codeArray[i];
+      dataObject.name = nameArray[i];
+      dataObject.value = valueArray[i];
+      newMaltreatmentDataObject.push(dataObject);
+    }
+    return newMaltreatmentDataObject;
+  };
+
+  // Variable Assignment Using Data Marshalling Methods.
+
+  const maltreatmentMeta = MALTREATMENT;
+  const geoid = selectedGeographicFeature;
+
+  const maltreatmentTypesList = getMaltreatmentTypeNames(maltreatmentTypes);
+
+  const maltreatmentTypesDataValues = getMaltreatmentSelectedValues(
+    data,
+    geography,
+    year,
+    geoid,
+    maltreatmentTypes
+  );
+
+  const maltreatmentTypesDataAggregate = getMaltreatmentAggregatedValue(
+    data,
+    geography,
+    year,
+    geoid,
+    maltreatmentTypes
+  );
+
+  const maltreatmentTypesDataObject = getMaltreatmentTypesDataObject(
+    maltreatmentTypes,
+    maltreatmentTypesList,
+    maltreatmentTypesDataValues
+  );
+
+  // Define Plotting Helper Methods.
+
+  const getBarVertTrace = (traceY, traceX, traceName, traceFillColor) => {
+    return {
+      y: [traceY],
+      x: [traceX],
+      name: traceName,
+      type: 'bar',
+      orientation: 'v',
+      marker: {
+        line: {
+          color: ['#111111'],
+          width: 1
+        },
+        color: [traceFillColor]
+      }
+    };
+  };
+
+  const getPlotDataVertBars = typesDataArray => {
+    const newPlotData = [];
+    for (let i = 0; i < typesDataArray.length; i += 1) {
+      const yData = typesDataArray[i].value;
+      const xData = typesDataArray[i].code;
+      const tName = typesDataArray[i].name;
+      const traceFillColor = plotCategoryColors[i];
+      const type = getBarVertTrace(yData, xData, tName, traceFillColor);
+      newPlotData.push(type);
+    }
+    return newPlotData;
+  };
+
+  // Assign Plot Variables.
+
+  const plotCategoryColors = [
```

Comment:
future refactor: could define these for the two different plots to do.

Reply:
Yeah, making a ticket to pull a number of methods out of the Plots into a plot helper module.